### PR TITLE
Add core dependency constraints to exclude bad core version

### PR DESCRIPTION
**Pull request by mattnichols**
# Summary of Changes

* Reorganize build.gradles to accomadate global constraint definitions that apply to the platform and the other sub-projects.
* Updating core dependencies to pull in newest core additions as well as exclude a bad version.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -34,8 +30,26 @@ allprojects {
...
+  } else {
     apply plugin: "java-library"
+  }
+
+  // Constraints for platform and all other projects
+  dependencies {
+    constraints {
+      api("com.mx.path-core:platform") {
+        version {
+          strictly "[1.4.0,2.0["
```
Comment (gerfboy) [inline build.gradle:45]: We talked about standardizing on [1.4.0,2.0).

Also, instead of a 'reject "1.4.0"' we could just do (1.4.0,2.0)

**Review by gerfboy — APPROVED** [review]
I left a comment on syntax, but what is here will work the same, so I'll approve.

File: build.gradle (hunk 1 above)
Comment (mattnichols) [reply build.gradle:45]: I'm rejecting it explicitly for clarity

Comment (mattnichols) [reply build.gradle:45]: This is also a bit experimental to see if this pattern will work for other projects. Right now, the constraints used by the projects and the sister platform project are defined separately, which seems gross.

Comment (gerfboy) [reply build.gradle:45]: 👍

**Pull request merged**